The new global config type aliases do not work well on `Option<>` types

Hi!

I've been trying out the new global config stuff on commit 576f6c129010b4f6e6445c06393cfcf2cfee30df.

I have the following test file:
```
use utoipa::ToSchema;
use chrono::{DateTime, Utc};

pub type Timestamp = DateTime<Utc>;

#[derive(ToSchema)]
pub struct S {
    d: Timestamp,
    d2: Option<Timestamp>,
}

fn main() {
    println!("Hello, world!");
}
```

and `build.rs`:
```
use utoipa_config::Config;

fn main() {
    Config::new()
        .alias_for("Timestamp", "DateTime<Utc>")
        .write_to_file();
}
````

However I am getting this error:
```
rror[E0277]: the trait bound `chrono::DateTime<Utc>: ToSchema` is not satisfied
 --> src/main.rs:9:16
  |
9 |     d2: Option<Timestamp>,
  |                ^^^^^^^^^ the trait `ToSchema` is not implemented for `chrono::DateTime<Utc>`
  |
  = help: the following other types implement trait `ToSchema`:
            &'a str
            ()
            S
            bool
            char
            f32
            f64
            i128
          and 13 others

error[E0277]: the trait bound `chrono::DateTime<Utc>: PartialSchema` is not satisfied
   --> src/main.rs:9:16
    |
9   |     d2: Option<Timestamp>,
    |                ^^^^^^^^^ the trait `ComposeSchema` is not implemented for `chrono::DateTime<Utc>`, which is required by `chrono::DateTime<Utc>: PartialSchema`
    |
    = help: the trait `ComposeSchema` is implemented for `S`
    = note: required for `chrono::DateTime<Utc>` to implement `PartialSchema`
note: required by a bound in `name`
   --> /Users/eran/.cargo/git/checkouts/utoipa-797003a51754f4fd/576f6c1/utoipa/src/lib.rs:402:21
    |
402 | pub trait ToSchema: PartialSchema {
    |                     ^^^^^^^^^^^^^ required by this bound in `ToSchema::name`
...
410 |     fn name() -> Cow<'static, str>;
    |        ---- required by a bound in this associated function

For more information about this error, try `rustc --explain E0277`.
error: could not compile `ut` (bin "ut") due to 2 previous errors
```

I know the build.rs/global config does get used because if I remove the `alias_for` line, then even `d: Timestamp` produces an error.

Thanks!

--- a/utoipa-config/CHANGELOG.md
+++ b/utoipa-config/CHANGELOG.md
@@ -6,6 +6,10 @@
 
 * Add global config for `utiopa` (https://github.com/juhaku/utoipa/pull/1048)
 
+### Fixed
+
+* Fix generic aliases (https://github.com/juhaku/utoipa/pull/1083)
+
 ### Changed
 
 * Fixed broken tests at `utoipa-config`

--- a/utoipa-config/config-test-crate/src/main.rs
+++ b/utoipa-config/config-test-crate/src/main.rs
@@ -15,6 +15,8 @@ struct AliasValues {
     my_value: bool,
 
     date: MyDateTime,
+
+    optional_date: Option<MyDateTime>,
 }
 
 #[allow(unused)]

--- a/utoipa-gen/CHANGELOG.md
+++ b/utoipa-gen/CHANGELOG.md
@@ -24,6 +24,7 @@
 
 ### Fixed
 
+* Fix generic aliases (https://github.com/juhaku/utoipa/pull/1083)
 * Fix nest path config struct name (https://github.com/juhaku/utoipa/pull/1081)
 * Fix `as` attribute path format (https://github.com/juhaku/utoipa/pull/1080)
 * Fix allow response `content_type` without schema (https://github.com/juhaku/utoipa/pull/1073)

--- a/utoipa-gen/src/component.rs
+++ b/utoipa-gen/src/component.rs
@@ -696,15 +696,20 @@ impl ComponentSchema {
                 {
                     features.push(Nullable::new().into());
                 }
+                let child = type_tree
+                    .children
+                    .as_ref()
+                    .expect("ComponentSchema generic container type should have children")
+                    .iter()
+                    .next()
+                    .expect("ComponentSchema generic container type should have 1 child");
+                let alias = child.get_alias_type()?;
+                let alias = alias.as_ref().map_try(TypeTree::from_type)?;
+                let child = alias.as_ref().unwrap_or(child);
+
                 let schema = ComponentSchema::new(ComponentSchemaProps {
                     container,
-                    type_tree: type_tree
-                        .children
-                        .as_ref()
-                        .expect("ComponentSchema generic container type should have children")
-                        .iter()
-                        .next()
-                        .expect("ComponentSchema generic container type should have 1 child"),
+                    type_tree: child,
                     features,
                     description,
                 })?;
@@ -713,15 +718,20 @@ impl ComponentSchema {
                 schema_references.extend(schema.schema_references);
             }
             Some(GenericType::Cow | GenericType::Box | GenericType::RefCell) => {
+                let child = type_tree
+                    .children
+                    .as_ref()
+                    .expect("ComponentSchema generic container type should have children")
+                    .iter()
+                    .next()
+                    .expect("ComponentSchema generic container type should have 1 child");
+                let alias = child.get_alias_type()?;
+                let alias = alias.as_ref().map_try(TypeTree::from_type)?;
+                let child = alias.as_ref().unwrap_or(child);
+
                 let schema = ComponentSchema::new(ComponentSchemaProps {
                     container,
-                    type_tree: type_tree
-                        .children
-                        .as_ref()
-                        .expect("ComponentSchema generic container type should have children")
-                        .iter()
-                        .next()
-                        .expect("ComponentSchema generic container type should have 1 child"),
+                    type_tree: child,
                     features,
                     description,
                 })?;
@@ -731,15 +741,20 @@ impl ComponentSchema {
             }
             #[cfg(feature = "rc_schema")]
             Some(GenericType::Arc) | Some(GenericType::Rc) => {
+                let child = type_tree
+                    .children
+                    .as_ref()
+                    .expect("ComponentSchema rc generic container type should have children")
+                    .iter()
+                    .next()
+                    .expect("ComponentSchema rc generic container type should have 1 child");
+                let alias = child.get_alias_type()?;
+                let alias = alias.as_ref().map_try(TypeTree::from_type)?;
+                let child = alias.as_ref().unwrap_or(child);
+
                 let schema = ComponentSchema::new(ComponentSchemaProps {
                     container,
-                    type_tree: type_tree
-                        .children
-                        .as_ref()
-                        .expect("ComponentSchema rc generic container type should have children")
-                        .iter()
-                        .next()
-                        .expect("ComponentSchema rc generic container type should have 1 child"),
+                    type_tree: child,
                     features,
                     description,
                 })?;
@@ -810,14 +825,19 @@ impl ComponentSchema {
                 // additionalProperties denoting the type
                 // maps have 2 child schemas and we are interested the second one of them
                 // which is used to determine the additional properties
+                let child = type_tree
+                    .children
+                    .as_ref()
+                    .expect("ComponentSchema Map type should have children")
+                    .get(1)
+                    .expect("ComponentSchema Map type should have 2 child");
+                let alias = child.get_alias_type()?;
+                let alias = alias.as_ref().map_try(TypeTree::from_type)?;
+                let child = alias.as_ref().unwrap_or(child);
+
                 let schema_property = ComponentSchema::new(ComponentSchemaProps {
                     container,
-                    type_tree: type_tree
-                        .children
-                        .as_ref()
-                        .expect("ComponentSchema Map type should have children")
-                        .get(1)
-                        .expect("ComponentSchema Map type should have 2 child"),
+                    type_tree: child,
                     features,
                     description: None,
                 })?;
@@ -882,6 +902,9 @@ impl ComponentSchema {
         } else {
             child
         };
+        let alias = child.get_alias_type()?;
+        let alias = alias.as_ref().map_try(TypeTree::from_type)?;
+        let child = alias.as_ref().unwrap_or(child);
 
         let component_schema = ComponentSchema::new(ComponentSchemaProps {
             container,

